dlpy/network.py

```diff
@@ -0,0 +1,1598 @@
+                                ' train the model first.')
+            print('NOTE: Model weights will be fetched from server')
+            model_weights = self.model_weights
+        else:
+            print('NOTE: Model weights will be loaded from csv.')
+            model_weights = pd.read_csv(model_weights)
+        model_table = self.conn.CASTable(**self.model_table)
+        onnx_model = sas_to_onnx(layers = self.layers,
+                                 model_table = model_table,
+                                 model_weights = model_weights)
+        file_name = self.model_name + '.onnx'
+        if path is None:
+            path = os.getcwd()
+
+        if not os.path.isdir(path):
+            os.makedirs(path)
+
+        file_name = os.path.join(path, file_name)
+
+        with open(file_name, 'wb') as f:
+            f.write(onnx_model.SerializeToString())
+
+        print('NOTE: ONNX model file saved successfully.')
+
+    def deploy(self, path, output_format = 'astore', model_weights = None, **kwargs):
+        """
+        Deploy the deep learning model to a data file
+
+        Parameters
+        ----------
+        path : string
+            Specifies the client-side path to store the model files.
+        output_format : string, optional
+            Specifies the format of the deployed model
+            Valid Values: astore, castable, or onnx
+            Default: astore
+        model_weights : string, optional
+            Specifies the client-side path to the csv file of the
+            model weights table.  Only effective when
+            output_format='onnx'.  If no csv file is specified when
+            deploying to ONNX, the weights will be fetched from the
+            CAS server.  This may take a long time to complete if
+            the size of model weights is large.
+
+        Notes
+        -----
+        Currently, this function supports sashdat, astore, and onnx formats.
+
+        More information about ONNX can be found at: https://onnx.ai/
+
+        DLPy supports ONNX version >= 1.3.0, and Opset version 8.
+
+        For ONNX format, currently supported layers are convo, pool,
+        fc, batchnorm, residual, concat, and detection.
+
+        If dropout is specified in the model, train the model using
+        inverted dropout, which can be specified in :class:`Optimizer`.
+        This will ensure the results are correct when running the
+        model during test phase.
+
+
+        """
+        if output_format.lower() == 'astore':
+            self.save_to_astore(path = path, **kwargs)
+        elif output_format.lower() in ('castable', 'table'):
+            self.save_to_table(path = path)
+        elif output_format.lower() == 'onnx':
+            self.save_to_onnx(path, model_weights = model_weights)
+        else:
+            raise DLPyError('output_format must be "astore", "castable", "table",'
+                            'or "onnx"')
+
+    def count_params(self):
```

Comment:
maybe we can put this inside compile or while building the model when we are traversing each layer